CRM/Financial/BAO/ExportFormat/IIF.php

```diff
@@ -351,7 +351,7 @@ public static function format($s, $type = 'string') {
     $s1 = str_replace(self::$SEPARATOR, '\t', $s);
     switch ($type) {
       case 'date':
-        $sout = date('Y/m/d', strtotime($s1));
+        $sout = CRM_Utils_Date::customFormat(date('Ymd', strtotime($s1)));
```

Comment:
Umm, if you want to use a date format, you will need to provide it. See https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/Date.php#L316
So, something like
$custom_format = value_from_setting(); // replace value_from_setting as appropropriate
CRM_Utils_Date::customFormat(date('Ymd', strtotime($s1)), $custom_format);